lib/ansible/modules/cloud/docker/docker_network.py

```diff
@@ -213,14 +214,10 @@ def __init__(self, client):
             self.absent()
 
     def get_existing_network(self):
-        networks = self.client.networks(names=[self.parameters.network_name])
-        # check if a user is trying to find network by its Id
-        if not networks:
-            networks = self.client.networks(ids=[self.parameters.network_name])
-        if not networks:
+        try:
+            return self.client.inspect_network(self.parameters.network_name)
```

Comment:
Does `inspect_network()` work both with the network's name and the network's ID?

Reply:
https://github.com/docker/docker-py/blob/master/docker/api/network.py#L189

docker-py assumes that a network's ID is being passed to it, but according to [1], both a network's name and network's ID is accepted on the endpoint being called in this inspect_network() function.


[1] : https://docs.docker.com/engine/api/v1.24/#35-networks

Reply:
I would interpret it that way too. That's something we should test in an integration test, though ;)

Reply:
Definitely! @felixfontein, I'm not familiar with integration testing with the Ansible codebase but I would love trying to work on a few tests.